### PR TITLE
refactor(code): Prevent checking for hyperspace capabilities while hyperspacing

### PR DESCRIPTION
**Pull request by Zitchas**
**Feature:**

## Feature Details
I noticed (via the additional flagship info that I have access to in Delta), that during the course of hyperjumping my ship AI is continuing to run a variety of checks, including calling PrepareForHyperspace that is trying to get my ship lined up for jumping, resulting in my ship receiving turn commands while jumping. Due to the fact its jumping it can't act on any of these, so nothing is visible in vanilla,  but they are happening nonetheless.

With the addition of an additional condition on this `else if` statement to check that the ship isn't currently hyperspacing, the game no longer runs this series of checks, nor issues any commands to try to continue lining up the ship after it has already begun the jump. 

## UI Screenshots
N/A

## Usage Examples
N/A

## Testing Done
Playing the game, noting that my ship is no longer trying to turn while hyperjumping. No unusual behavior was noted in NPC ships while flying around.

### Automated Tests Added
N/A

## Performance Impact
This optimization prevents a number of checks and calls from happening when they are useless, so it should improve performance, albeit by a very minute amount.

### CODE REVIEW

File: source/AI.cpp [hunk 1]
```diff
@@ -4124,7 +4124,7 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 		if(Stop(ship, command))
 			autoPilot.Clear(Command::STOP);
 	}
-	else if(autoPilot.Has(Command::JUMP | Command::FLEET_JUMP))
+	else if(autoPilot.Has((Command::JUMP | Command::FLEET_JUMP)) && !ship.IsHyperspacing())
```
Comment (warp-core) [inline source/AI.cpp:4127]:
```suggestion
	else if(autoPilot.Has(Command::JUMP | Command::FLEET_JUMP) && !ship.IsHyperspacing())
```

**Comment by Zitchas** [comment]
OK, so the restriction will need to be fine tuned a bit, then. Thanks!

**Comment by vitalchip** [comment]
Use IsEnteringHyperspace() rather than IsHyperspacing() as it only returns true for the first half of the jump, so as soon as you arrive in the destination system the checks can be made for any onward travel.

The autopilot holds the jump command from the moment you press 'J' until the jump is completed or the autopilot cancelled by pressing a movement key.

**Comment by Zitchas** [comment]
Currently replicating this change on a local copy of Delta for testing purposes, as there's nothing in vanilla ES that indicates whether or not this change is actually doing what it is intended to do.

**Comment by Zitchas** [comment]
OK,  yes, this change works as intended, mostly. Sometimes it starts trying to turn after it crosses into the next system before it has completed the jump, so this doesn't fix *all* the useless activity; but it does fix the larger portion. From an in-game standpoint, seeing that the ship is trying to do turns or whatnot after having transitioned to the destination system but before the HD has let go does kind of make sense: It could be lored to be re-establishing proper flight path or something. 

Anyway, I'm happy with it.

**Comment by warp-core** [comment]
> Sometimes it starts trying to turn after it crosses into the next system before it has completed the jump

So, you've replaced one behaviour that you don't understand with another one?

**Comment by Zitchas** [comment]
No, it's acting exactly as described (See VitalChip's description). It's replacing something active for the whole jump with something active for only the first half, in order to allow the desireable part you mentioned not wanting to block in the second half.

**Pull request merged**